tests/samples_test.py

```diff
@@ -0,0 +1,359 @@
+    "NestedTrigonometric [0]",
+    "NestedTrigonometric [1]",
+    "NestedTrigonometric [2]",
+]
+
+
+def test_clenshaw_initial():
+    """test default properties"""
+    f = smolyay.samples.ClenshawCurtisPointSet([-2, 1], 7)
+    assert numpy.array_equal(f.domain, [-2, 1])
+    assert f.degree == 7
+    assert isinstance(f.degree, int)
+    f.degree = float(5)
+    assert f.degree == 5
+    assert isinstance(f.degree, int)
+
+
+def test_trig_initial():
+    """test default properties"""
+    f = smolyay.samples.TrigonometricPointSet([0, 4 * numpy.pi], 7)
+    assert numpy.array_equal(f.domain, [0, 4 * numpy.pi])
+    assert f.frequency == 7
+    assert isinstance(f.frequency, int)
+    f.frequency = float(5)
+    assert f.frequency == 5
+    assert isinstance(f.frequency, int)
+
+
+@pytest.mark.parametrize(
+    "nested_samples",
+    [
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_nested_initial(nested_samples):
+    """test default properties"""
+    f = nested_samples([-10, 10], 7)
+    assert numpy.array_equal(f.domain, [-10, 10])
+    assert f.max_level == 7
+    assert isinstance(f.max_level, int)
+    f.max_level = float(5)
+    assert f.max_level == 5
+    assert isinstance(f.max_level, int)
+
+
+@pytest.mark.parametrize(
+    "samples",
+    [
+        smolyay.samples.ClenshawCurtisPointSet,
+        smolyay.samples.TrigonometricPointSet,
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "ClenshawCurtis",
+        "Trigonometric",
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_domain_error(samples):
+    """test error given invalid domain"""
+    with pytest.raises(TypeError):
+        samples([-10, 10, 20], 7)
+    with pytest.raises(TypeError):
+        samples([[-10, 10], [-10, 10]], 7)
+
+
+@pytest.mark.parametrize("samples,points", sample_points_answers, ids=sample_points_ids)
+def test_generate_points(samples, points):
+    """test the points of initialized UnidimensionalPointSet"""
+    assert len(samples) == len(points)
+    assert numpy.allclose(samples.points, points, atol=1e-10)
+
+    # test for different domain and that points update after changing domain
+    new_points = numpy.array(points) * 5 + 10
+    new_domain = samples.domain * 5 + 10
+    samples.domain = new_domain
+    assert numpy.array_equal(samples.domain, new_domain)
+    assert numpy.allclose(samples.points, new_points, atol=1e-10)
+    assert numpy.allclose(samples[:], new_points, atol=1e-10)
+    for p, a in zip(samples, new_points):
+        assert p == pytest.approx(a)
```

Comment:
I would add an explicit single element check, and then your for-loop (testing that it works as an iterator) can be written more compactly by turning it into a list.
```suggestion
    assert samples[0] == pytest.approx(new_points[0])
    assert numpy.allclose(samples[:], new_points, atol=1e-10)
    assert numpy.allclose(list(samples), new_points, atol=1e-10)
```